Dockerfile

```diff
@@ -20,9 +40,27 @@ COPY ./api/alembic /app/alembic
 COPY ./api/alembic.ini /app
 # Copy pre-start.sh (it will be run on startup):
 COPY ./api/prestart.sh /app
+COPY ./api/start.sh /app
 
 # The fastapi docker image defaults to port 80, but openshift doesn't allow non-root users port 80.
 EXPOSE 8080
 
 # Set the classpath to include copied libs
 ENV CLASSPATH=/app/libs/REDapp_Lib.jar:/app/libs/WTime.jar:/app/libs/hss-java.jar:${CLASSPATH}
+# Tell poetry where to find the cache
+ENV POETRY_CACHE_DIR="/home/${USERNAME}/.cache/pypoetry"
+# Put poetry on the path
+ENV PATH="/home/${USERNAME}/.local/bin:${PATH}"
+
+# root user please
+USER 0
+# We don't know what user poetry is going to run as, so we give everyone write access directories
+# in the app folder. We need write access for .pyc files to be created. .pyc files are good,
+# they speed up python.
+RUN chmod a+w $(find /app/app -type d)
+
+# Openshift runs with a random non-root user, so switching our user to 1001 allows us
```

Comment:
If you run docker locally, and you switched to some user in your Dockerfile, it will be that user when you run your container. In Openshift, you get a random user.